crates/biome_js_analyze/src/semantic_analyzers/correctness/no_const_assign.rs

```diff
@@ -113,4 +116,58 @@ impl Rule for NoConstAssign {
             ),
         )
     }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let model = ctx.model();
+        let mut mutation = ctx.root().begin();
+
+        let declared_binding = model.binding(node)?;
+
+        if let Some(possible_declarator) = declared_binding.syntax().ancestors().find(|node| {
+            !AnyJsObjectBindingPatternMember::can_cast(node.kind())
+                && !JsObjectBindingPatternPropertyList::can_cast(node.kind())
+                && !AnyJsArrayBindingPatternElement::can_cast(node.kind())
+                && !JsArrayBindingPatternElementList::can_cast(node.kind())
+                && !JsIdentifierBinding::can_cast(node.kind())
+        }) {
```

Comment:
You can use `declaration` method (I didn't test the code):
```suggestion
        if let AnyJsBindingDeclaration::JsVariableDeclarator(possible_declarator) = declared_binding.declaration()? {
```

Reply:
Hm! no such method on `binding`.

Reply:
Oh yes, You need to first call `declared_binding.tree()` that returns a `AnyJsIdentifierBinding`.
You also need to import `biome_js_syntax::binding_ext::AnyJsIdentifierBinding`

```suggestion
        if let AnyJsBindingDeclaration::JsVariableDeclarator(possible_declarator) = declared_binding.tree().declaration()? {
```